File: backend/app/schemas/data_management.py
from datetime import datetime

from pydantic import BaseModel


class StorageInfoResponse(BaseModel):
    data_dir: str
    database_path: str
    database_exists: bool
    database_size_bytes: int
    backups_dir: str
    backup_count: int


class BackupInfoResponse(BaseModel):
    filename: str
    path: str
    size_bytes: int
    created_at: datetime


class BackupCreateResponse(BaseModel):
    message: str
    backup: BackupInfoResponse

File: frontend/src/api/dataManagement.ts
import { apiClient } from "./client";

export type StorageInfo = {
  data_dir: string;
  database_path: string;
  database_exists: boolean;
  database_size_bytes: number;
  backups_dir: string;
  backup_count: number;
};

export type BackupInfo = {
  filename: string;
  path: string;
  size_bytes: number;
  created_at: string;
};

export type BackupCreateResponse = {
  message: string;
  backup: BackupInfo;
};

export async function fetchStorageInfo(): Promise<StorageInfo> {
  const res = await apiClient.get("/data/storage");
  return res.data;
}

export async function fetchBackups(): Promise<BackupInfo[]> {
  const res = await apiClient.get("/data/backups");
  return res.data;
}

export async function createBackup(): Promise<BackupCreateResponse> {
  const res = await apiClient.post("/data/backups");
  return res.data;
}
